back/engines/commercial/analysis/app/services/analysis/inputs_finder.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module Analysis
+  class InputsFinder
+    MAX_PER_PAGE = 100
+
+    attr_reader :analysis, :params
+
+    def initialize(analysis, params = {})
+      @analysis = analysis
+      @params = params
+    end
+
+    def execute
+      inputs = analysis.inputs
+
+      inputs = filter_published_at(inputs)
+      inputs = filter_reactions(inputs)
+      inputs = filter_comments(inputs)
+      inputs = filter_votes(inputs)
+      inputs = filter_custom_field_in(inputs)
+      inputs = filter_custom_field_range(inputs)
+      search(inputs)
+    end
+
+    private
+
+    def filter_published_at(inputs)
+      scope = inputs
+
+      scope = scope.where('published_at >= ?', params[:published_at_from]) if params[:published_at_from]
```

Comment:
Not sure I understand what you mean. You mean when `params[:publsihed_at_from]` is defined, but e.g. `nil`?
In that case this will just be skipped?

Reply:
You're totally right. I just confused with declarations (that is if the left-hand side is not defined yet).